src/components/drop-down/drop-down.tsx

```diff
@@ -70,15 +62,62 @@ export class DropDown extends React.Component<DropDownProps, {}> {
         });
 
         return (
-            <div data-automation-id="DROP_DOWN" {...rootProps}>
-                <DropDownInput selectedItem={this.props.selectedItem} onClick={this.props.onInputClick} />
-                <DropDownList
-                    selectedItem={this.props.selectedItem}
-                    items={this.props.items}
-                    open={!!this.props.open}
-                    onItemClick={this.onItemClick}
-                />
+            <div
+                {...rootProps}
+                onKeyDown={this.onKeyDown}
+                onFocus={this.onFocus}
+                tabIndex={this.props.tabIndex}
+                ref={dropdown => {
+                    if (!this.state.dropdown) {
+                        this.setState({dropdown});
+                    }
+                }}
+            >
+                <div data-automation-id="DROP_DOWN_INPUT" onClick={this.onInputClick} className="drop-down-input">
+                    <span className="label">{this.props.value!}</span>
+                    {this.props.toggleIcon
+                        ? this.props.toggleIcon
+                        : <CaretDown className="caret" />
+                    }
+                </div>
+                <Popup open={!!this.props.open && !this.props.disabled} anchor={this.state.dropdown}>
+                    <div className="root">
+                        <SelectionList
+                            data-automation-id="DROP_DOWN_LIST"
+                            className="drop-down-list"
+                            value={this.props.value}
+                            onChange={this.onItemClick!}
+                            dataSource={this.props.children}
```

Comment:
`dataSource={this.props.children}` makes no sense. what is the type of dataSource?